### PR TITLE
Use `__info__` callback directly

### PR DESCRIPTION
**Pull request by josevalim**
@lukaszsamson, what do you think about this change?

### CODE REVIEW

**Comment by josevalim** [comment]
If you agree, I will merge and backport both of our commits.

File: lib/elixir/lib/macro/env.ex [hunk 1]
```diff
@@ -351,7 +351,9 @@ defmodule Macro.Env do
 
     * `:info_callback` - a function to use instead of `c:Module.__info__/1`.
       The function will be invoked with `:functions` or `:macros` argument.
-      It has to return a list of `{function, arity}` key value pairs
+      It has to return a list of `{function, arity}` key value pairs.
+      If it fails, it defaults to using module metadata based on `module_info`
+      and `behaviour_info`.
```
Comment (lukaszsamson) [inline lib/elixir/lib/macro/env.ex:356]: `behaviour_info` is not used here, only filtered out from the exports list

**Comment by lukaszsamson** [comment]
I'm OK with that version as well

File: lib/elixir/lib/macro/env.ex (hunk 1 above)
Comment (josevalim) [inline lib/elixir/lib/macro/env.ex:356]:
```suggestion
      If it fails, it defaults to using module metadata based on `module_info/1`.
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**